vm/src/version.rs

```diff
@@ -42,16 +46,28 @@ pub fn get_version_number() -> String {
 
 pub fn get_compiler() -> String {
     let rustc_version = rustc_version_runtime::version_meta();
-    format!("rustc {}", rustc_version.semver)
+    format!("\n[rustc {}]", rustc_version.semver)
 }
 
-pub fn get_build_info() -> (String, String) {
-    let git_hash = get_git_revision();
+pub fn get_build_info() -> String {
     // See: https://reproducible-builds.org/docs/timestamps/
-    let git_timestamp = option_env!("RUSTPYTHON_GIT_TIMESTAMP")
-        .unwrap_or("")
-        .to_string();
-    (git_hash, git_timestamp)
+    let git_revision = get_git_revision();
+    let separator = if git_revision.is_empty() { "" } else { ":" };
+
+    let git_identifier = get_git_identifier();
+
+    format!(
+        "{id}{sep}{revision}, {date:.20}, {time:.9}",
+        id = if git_identifier.is_empty() {
```

Comment:
You might want to pull this out of the `format!` macro, and first calculate `id` as a variable using `let id = ...` before calling `format!`.

Reply:
Thank you for suggesting, but I didn't want to make another variable for formatting.